crates/utils/src/request.rs

```diff
@@ -107,37 +138,43 @@ pub(crate) async fn fetch_pictrs(
       .map_err(|e| RecvError(e.to_string()))?;
 
     if response.msg == "ok" {
-      Ok(Some(response))
+      Ok(response)
     } else {
       Err(anyhow!("{}", &response.msg).into())
     }
   } else {
-    Ok(None)
+    Err(anyhow!("pictrs_url not set up in config").into())
   }
 }
 
-pub async fn fetch_iframely_and_pictrs_data(
+/// Both are options, since the URL might be either an html page, or an image
+pub async fn fetch_site_metadata_and_pictrs_data(
```

Comment:
Imo the thumbnail is also part of metadata, so no need to mention pictrs explicitly here. This method should definitely log errors (under verbose). A more detailed description of what it does would also be nice.

Reply:
There's already a `fetch_site_metadata` function, which refers specifically to this `SiteMetadata` struct I just added. I can change the name to `fetch_site_data` tho.